include/w1.h

```diff
@@ -0,0 +1,370 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for the 1-Wire drivers.
+ */
+
+/*
+ * Copyright (c) 2018 Roman Tataurov <diytronic@yandex.ru>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __DRIVERS_W1_H
+#define __DRIVERS_W1_H
+
+/**
+ * @brief 1-Wire Interface
+ * @defgroup w1_interface 1-Wire Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/*
+ * The following #defines are used to configure the 1-Wire controller.
+ */
+#define W1_SEARCH		0xF0
+#define W1_ALARM_SEARCH		0xEC
+#define W1_CONVERT_TEMP		0x44
+#define W1_SKIP_ROM		0xCC
+#define W1_COPY_SCRATCHPAD	0x48
+#define W1_WRITE_SCRATCHPAD	0x4E
+#define W1_READ_SCRATCHPAD	0xBE
+#define W1_READ_ROM		0x33
+#define W1_READ_PSUPPLY		0xB4
+#define W1_MATCH_ROM		0x55
+#define W1_RESUME_CMD		0xA5
+
+#define W1_UUID_INIT(value...) (w1_reg_num *){ value }
+
+/*
+ * Precalculated CRC table
+ */
+static u8_t w1_crc8_table[] = {
```

Comment:
static variable in the header?? That will create that variable in RAM for every file that includes `w1.h`. Also it should be const.

Reply:
Hmm .... what about include guard in this header file? It should guarantee what file included only one time.

Reply:
Hmm ... clarified it - seems you are right - will fix it